server/router.js

```diff
@@ -0,0 +1,163 @@
+'use strict';
+
+const express = require('express');
+const bodyParser = require('body-parser');
+const methodOverride = require('method-override');
+const db = require('./db');
+
+const router = express.Router();
+
+let movieCount;
+
+router.use(bodyParser.urlencoded({ extended: true }));
+router.use(methodOverride('_method'));
+
+// Load Landing page
+router.get('/', (req, res) => {
+  res.render('landing', {
+    pageId: 'landing',
+    title: 'Home',
+  });
+});
+
+// Register route
+router.get('/register', (req, res) => {
+  res.render('./register');
+});
+
+// Login Route
+router.get('/login', (req, res) => {
+  try {
+    res.render('./login');
+  } catch (next) {
+    res.render('error');
+  }
+});
+
+// Render the movies page
+router.get('/movies', (req, res, next) => {
+  db.getAllMovies()
+    .then((movies) => {
+      res.render('movies', {
+        pageId: 'movies',
+        title: 'Movies',
+        movies,
+      });
+      movieCount = movies.length;
```

Comment:
I would avoid state persisting between requests but instead recalculate it each time.